File: schemas/hr-payroll/mongodb/earning_types.js
// earning_types: Classification of earning categories (regular, overtime, bonus, etc.) with tax flags.
// See README.md for full design rationale and field documentation.

const mongoose = require("mongoose");

const earningTypesSchema = new mongoose.Schema(
  {
    name: { type: String, required: true },
    code: { type: String, required: true },
    category: {
      type: String,
      enum: ["regular", "overtime", "bonus", "commission", "reimbursement", "other"],
      required: true,
    },
    is_taxable: { type: Boolean, required: true, default: true },
    is_active: { type: Boolean, required: true, default: true },
    description: { type: String, default: null },
  },
  {
    timestamps: { createdAt: "created_at", updatedAt: "updated_at" },
  }
);

earningTypesSchema.index({ code: 1 }, { unique: true });
earningTypesSchema.index({ category: 1 });
earningTypesSchema.index({ is_active: 1 });

module.exports = mongoose.model("EarningType", earningTypesSchema);

File: schemas/hr-payroll/spacetimedb/earning_types.rs
// earning_types: Catalog of earning categories (regular, overtime, bonus, etc.) used to classify payroll line items.
// See README.md for full design rationale.

use spacetimedb::{SpacetimeType, Timestamp};

// type: String
#[derive(SpacetimeType, Clone)]
pub enum EarningCategory {
    Regular,
    Overtime,
    Bonus,
    Commission,
    Reimbursement,
    Other,
}

#[spacetimedb::table(name = earning_types, public)]
pub struct EarningType {
    #[primary_key]
    pub id: String, // UUID

    pub name: String,

    #[unique]
    pub code: String,

    #[index(btree)]
    pub category: EarningCategory,

    pub is_taxable: bool,   // default: true

    #[index(btree)]
    pub is_active: bool,    // default: true

    pub description: Option<String>,

    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
